### PR TITLE
Move MessageLogger default configuration to be part of Process

### LINKED ISSUE #272: Move MessageLogger default configuration to be part of Process
Motivated by discussion in https://github.com/cms-sw/cmssw/pull/35117

Done in https://github.com/cms-sw/cmssw/pull/35298

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

MessageLogger is one of the Services always available in the framework. Following the improved configuration validation already in python, this PR suggests to move the MessageLogger default configuration to be available in the `cms.Process` without any extra action. This change allows using `ProcessModifier`s in any cff/cfi fragment to e.g. add new categories for which Info messages would be printed out (see e.g. discussion in https://github.com/cms-sw/cmssw/pull/35117).

Resolves https://github.com/makortel/framework/issues/272

#### PR validation:

Framework unit tests run.

### CODE REVIEW

**Comment by makortel** [comment]
@Dr15Jones What do you think?

File: FWCore/ParameterSet/python/Config.py [hunk 1]
```diff
@@ -137,6 +138,7 @@ def __init__(self,name,*Mods):
         self.options = Process.defaultOptions_()
         self.maxEvents = Process.defaultMaxEvents_()
         self.maxLuminosityBlocks = Process.defaultMaxLuminosityBlocks_()
+        self.MessageLogger = MessageLogger # intentionally not cloned
```
Comment (makortel) [inline FWCore/ParameterSet/python/Config.py:141]: Not cloning here means that in the following
```py
process = cms.Process("Foo")
process.MessageLogger.SomeCategory = cms.untracked.PSet()
process.load('FWCore.MessageService.MessageLogger_cfi')
```
the `process.load()` does not override the settings done before it.

**Comment by makortel** [comment]
@cmsbuild, please test

File: FWCore/MessageService/python/MessageLogger_cfi.py [hunk 2]
```diff
@@ -1,91 +1 @@
...
-        threshold = cms.untracked.string('INFO')
-    ),
-    files = cms.untracked.PSet(
-        allowAnyLabel_ = _file_destination
-    ),
-    allowAnyLabel_ = _category
-)
-
-
+from FWCore.ParameterSet.MessageLogger import MessageLogger
```
Comment (fwyzard) [inline FWCore/MessageService/python/MessageLogger_cfi.py:1]: maybe, apply similar changes also to [FWCore/MessageLogger/python/MessageLogger_cfi.py](https://github.com/cms-sw/cmssw/blob/master/FWCore/MessageLogger/python/MessageLogger_cfi.py)

File: FWCore/ParameterSet/python/Config.py (hunk 1 above)
Comment (Dr15Jones) [reply FWCore/ParameterSet/python/Config.py:141]: Which, I'm assuming, is the desired behavior (to allow independent modifications of the MessageLogger in different packages).

File: FWCore/ParameterSet/python/Config.py [hunk 3]
```diff
@@ -1479,6 +1481,8 @@ def __init__(self,process, SelectEvents = untracked.PSet(), outputCommands = unt
         self.__process = process
         self.__SelectEvents = SelectEvents
         self.__outputCommands = outputCommands
+        # Need to remove MessageLogger from the subprocess now that MessageLogger is always present
+        del self.__process.MessageLogger
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/python/Config.py:1485]: Are the MessageLogger objects guaranteed to be the same? [Hopefully] If they are not, wouldn't we want to 'forward' extra categories to the parent?

File: FWCore/ParameterSet/python/Config.py [hunk 4]
```diff
@@ -2842,7 +2941,12 @@ def testSubProcess(self):
 process.addSubProcess(cms.SubProcess(process = childProcess, SelectEvents = cms.untracked.PSet(
 ), outputCommands = cms.untracked.vstring()))"""
             equalD = equalD.replace("parentProcess","parentProcess"+str(hash(process.subProcesses_()[0])))
-            self.assertEqual(_lineDiff(d,Process('Parent').dumpPython()+Process('Child').dumpPython()),equalD)
+            messageLoggerSplit = 'process.MessageLogger = cms.Service'
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/python/Config.py:2944]: How about a comment explaining why this was done?

**Comment by Dr15Jones** [comment]
I'm done with my initial review.

File: FWCore/MessageService/python/MessageLogger_cfi.py (hunk 2 above)
Comment (makortel) [reply FWCore/MessageService/python/MessageLogger_cfi.py:1]: Done, thanks.

File: FWCore/ParameterSet/python/Config.py (hunk 4 above)
Comment (makortel) [reply FWCore/ParameterSet/python/Config.py:2955]: Added comment.

**Comment by perrotta** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
@makortel now  #35117 is merged. Perhaps, when the corresponding IB will become available you could take into account the [comment](https://github.com/cms-sw/cmssw/pull/35117#issuecomment-922521936) of @fwyzard already here

File: FWCore/ParameterSet/python/Config.py (hunk 3 above)
Comment (makortel) [reply FWCore/ParameterSet/python/Config.py:1495]: I added a printout of a similar warning if the `process.MessageLogger` objects differ between the parent Process and the SubProcess. If the objects are the same, all changes done via the SubProcess exist automatically already in the parent Process too.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@perrotta 
> now #35117 is merged. Perhaps, when the corresponding IB will become available you could take into account the [comment](https://github.com/cms-sw/cmssw/pull/35117#issuecomment-922521936) already here

I'd prefer to do cleanup in separate PRs, starting from ConfigBuilder (`git grep "FWCore\.Message.*\.MessageLogger" | wc -l` gives 2113).

**Comment by makortel** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**